File: Object Pool/Golang/ThreadPool.go
package main

type ThreadPool struct {
	threads []*Thread
	counter int
}

func NewThreadPool() *ThreadPool {
	threadPool := ThreadPool{}
	threadPool.threads = make([]*Thread, 0)
	threadPool.counter = 2

	for i := 1; i <= 2; i++ {
		thread := &Thread{
			number: i,
		}

		threadPool.threads = append(threadPool.threads, thread)
	}

	return &threadPool
}

func (threadPool *ThreadPool) get() *Thread  {
	var thread *Thread
	if len(threadPool.threads) > 0 {
		thread = threadPool.threads[0]
		threadPool.threads = threadPool.threads[1:]

	} else {
		threadPool.counter += 1
		thread = &Thread{
			number: threadPool.counter,
		}
	}

	return thread
}

func (threadPool *ThreadPool) release(thread *Thread) {
	threadPool.threads = append(threadPool.threads, thread)
}

File: Object Pool/Python 3/ThreadPool.py
from Thread import Thread

class ThreadPool:
  def __init__(self):
    self.threads = []
    self.thread_count = 2

    for i in range(1, self.thread_count + 1):
      self.threads.append(Thread(i))
  
  def get(self):
    thread = None

    if len(self.threads) > 0:
      thread = self.threads[0]
      del self.threads[0]
    else:
      self.thread_count += 1
      thread = Thread(self.thread_count)
    
    return thread
  
  def release(self, thread):
    self.threads.append(thread)